kevm-pyk/src/kevm_pyk/kproj/evm-semantics/serialization.md

```diff
@@ -59,10 +59,10 @@ Address/Hash Helpers
     rule [#newAddr]:        #newAddr(ACCT, NONCE) => #addr(#parseHexWord(Keccak256bytes(#rlpEncode([#addrBytes(ACCT), NONCE]))))                                                          [concrete]
     rule [#newAddrCreate2]: #newAddr(ACCT, SALT, INITCODE) => #addr(#parseHexWord(Keccak256bytes(b"\xff" +Bytes #addrBytes(ACCT) +Bytes #wordBytes(SALT) +Bytes #parseByteStack(Keccak256bytes(INITCODE))))) [concrete]
 
-    syntax Account ::= #sender ( TxData , Int , Bytes , Bytes ) [function, klabel(#senderTxData)]
-                     | #sender ( Bytes  , Int , Bytes , Bytes ) [function, klabel(#senderAux)   ]
-                     | #sender ( Bytes )                        [function, klabel(#senderAux2)  ]
- // -----------------------------------------------------------------------------------------------
+    syntax Account ::= #sender ( TxData , Int , Bytes , Bytes ) [function, klabel(#senderTxData)     ]
+                     | #sender ( Bytes  , Int , Bytes , Bytes ) [function, klabel(#senderAux), total ]
+                     | #sender ( Bytes )                        [function, klabel(#senderAux2), total]
+ // -------------------------------------------------------------------------------------------------
```

Comment:
Is `#sender` actually total? I don't think so. It uses, for example, `ECDSARecover`, which is not total: https://github.com/runtimeverification/blockchain-k-plugin/blob/master/plugin/krypto.md#ecdsa-functions

Reply:
My mistake.

`#sender` uses several other non-total functions:

- `#parseHexWord`
- `#parseByteStack`
- `#hashTxData`

I think only the second `#sender` can be marked as `total`, because it invokes the 3rd `#sender` on arguments that we should be able to guarantee that are parseable by `#parseHexWord`.

Reply:
OK, removing the attribute from `#sender ( Bytes )`. The goal was to mark `#ecrec`  as total, which we can keep.